### ISSUE TITLE
We must be doing something wrong...

### ISSUE THREAD

**Issue by bhaugen** [valueflows/valueflows#92]
@edwardwest in Hylo said,
> I want to make sure that @Mikey Williams in particular gets a sense for CEPTR and it's aims, as it may simplify some of what Value Flows is proposing.

https://www.hylo.com/p/11921

Which implies to me that we must be proposing some unnecessary complexity.

@ahdinosaur ? @edwardwest did you get the ping? Can you explain what unnecessary complexity we are imposing? Doesn't seem like a good thing...

**Comment by edwardwest** [comment]
**Please forgive me if this note betrays my lack of understanding**: 

Here's what I was pointing to: 

Value Flows creates simple harmonized taxonomies, but the complexity it introduces is in having them be adopted widely by disparate projects. 

CEPTR (in theory) allows disparate projects to share data, and simplifies their ability to interact with a layer of technical complexity. 

Does the complexity introduced by CEPTR at a technical level make that bridge between projects function work more simply? 

What if anything can they gain from working together in some way?

**Comment by bhaugen** [comment]
@edwardwest thanks for responding. Don't worry about your understanding. We're all still figuring everything out. But here's how I _think_ things will evolve.

@fosterlynn and I work with existing groups on their existing problems that fit within our focus, which is economic networks. For example, we are currently members of the Madison WI Mutual Aid Network (MAN). They use two different systems: a mutual credit system that Matthew Slater developed, and Wezer, a derivative of Odoo, which is a derivative of Open ERP. They want to intertrade between both of those systems, responding to offers and wants and transferring hour-based currencies between accounts.

To the extent we work on that, we would propose the simplest thing that could possibly work for them, keeping everything we are doing in value flows in the back of our minds, but not force-fitting it into the situation. (In real life, I think Matthew will do all the work on the Mutual Credit side and be the biggest influence on the intertrading with Wezer, too. We'll mostly be along for the ride.)

But we will learn from that, and bring the learning back into the value flows project.

Then the Bergnek South Africa community will become a MAN, and they also plan to use our NRP software for a brick-production project. In that case, we will be more directly involved in the coding, but will take the same approach: make it work. And then take the learning back here.

Value flows is a summation and generalization project.

I would also be very interested in trying Ceptr in a real project, but don't want to have a hard dependency on Ceptr or anything else except the Web at this stage.

Does that all make sense to you?  This is pretty different from the way most software businesses work as they develop products, but we are not a business, and we don't have a product.

**Comment by elf-pavlik** [comment]
> CEPTR (in theory) allows disparate projects to share data, and simplifies their ability to interact with a layer of technical complexity.

@edwardwest  You can already experiment with using very small set of vocabulary terms currently included in ValueFlows. How can you in practice use CEPTR in your current codebase powering Hylo as of today?

**Comment by gcassel** [comment]
I don't understand CEPTR well enough to 'get' all of its potential, but I currently perceive value flows as a more versatile and flexible concept than any technical specification could be.  This does depend on designers' attitudes: if value flows vocabs were strictly prescribed as specifications, they could become as brittle and ultimately clunky as any tech dependencies.

**Comment by ahdinosaur** [comment]
@bhaugen: yeah, @edwardwest reached out to me regarding this, i'm just over capacity at the moment with other work to be able to contribute enough, i'm really glad that y'all are continuing to be active.

> @edwardwest says: Value Flows creates simple harmonized taxonomies, but the complexity it introduces is in having them be adopted widely by disparate projects.

what attracted me to the work of @bhaugen and @fosterlynn is that it's an _abstract_ vocabulary, in that it only describes the simplest possible set of verbs and nouns that are necessary in order for disparate projects to define _concrete_ vocabularies. we plan to do this using the existing RDF-style Linked Open Data literature and tooling.

in my opinion, us defining the _abstract_ so local communities can define the _concrete_ is one of the fundamental design principle of Value Flows, which separates it from most other Linked Open Data projects i'm aware of.

prior discussion:
- https://github.com/valueflows/valueflows/issues/63#issuecomment-148227341
- https://github.com/valueflows/agent/issues/44#issuecomment-147254440
- https://github.com/valueflows/agent/issues/26#issuecomment-143505501

**Comment by gcassel** [comment]
> what attracted me to the work of @bhaugen and @fosterlynn is that it's an abstract vocabulary, in that it only describes the simplest possible set of verbs and nouns that are necessary in order for disparate projects to define concrete vocabularies.

^ thank you Mikey. :)
Even if only a small group of people initially see the value in value flows, and develop it further, I believe that the current approach can scale to include diverse feedback with few if any 'deep adjustments'.

**Comment by elf-pavlik** [comment]
example from hylo-node:
https://github.com/Hylozoic/hylo-node/blob/master/api/models/Membership.js

Currently only hylo-frontend can understand this data model, if instead it used shared vocabulary, for example http://www.w3.org/TR/vocab-org/ (which we also evaluate as part of VF work). Any system which also implements it could understand data provided by services powered by hylo-node.

I see such common schemas, as a small simple change to what most people already do, but this change gives us a big interoperability gains.
I also have great appreciation to Eric, Arthur and others working on CEPTR, but it seems to me requiring a radical changes to what most people do today. I just can't stop thinking of other ambitious endeavors which try to propose pioneering new way of doing the web e.g. https://github.com/tav/ampify At the same time wondering if people working on them took the time first to understand current architecture of The Web http://www.w3.org/TR/webarch/

**Comment by bhaugen** [comment]
To add to what I wrote about about how I think things will evolve: that was only one direction, that of interconnecting current apps that people are using in the same situations.

The other, possibly more interesting direction, is developing new tools where the value flows vocab and protocols are "native", and composing systems out of those vf-native apps. 
That was the original idea of the Open Apps Ecosystem. The current examples of this direction are [Holodex](https://github.com/open-app/holodex), [Personator](https://github.com/holodex/personator), and the [Linked Data Browser](https://github.com/valueflows/linked-data-browser).

Following this direction, we will need to figure out the sweet spots for component size and shape for Value Flows Open Apps. If we do that well, many people should be able to create variations on any of the components, and they should be able to work together with any of the other variations. (I know, I know, lots of devils in those details, but still...)

**Comment by gcassel** [comment]
^ I think those last two comments are brilliant.
